client/components/webpack-build-monitor/index.tsx

```diff
@@ -0,0 +1,222 @@
+/**
+ * @see https://webpack.js.org/api/module-variables/#__webpack_hash__-webpack-specific
+ */
+declare const __webpack_hash__: string;
+
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import debugFactory from 'debug';
+import io from 'socket.io-client';
+
+/**
+ * Internal dependencies
+ */
+import Spinner from 'components/spinner';
+import { IntervalHandle, TimestampMS } from 'client/types';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const CONNECTION_TIMEOUT = 20 * 1000;
+const debug = debugFactory( 'calypso:webpack-build-monitor' );
+
+const WebpackBuildMonitor: FunctionComponent = () => {
+	const [ lastHash, setLastHash ] = useState( __webpack_hash__ );
+
+	// Webpack
+	const [ isWebpackConnected, setIsWebpackConnected ] = useState( false );
```

Comment:
I'm not sure if I'd have these many different state hooks. They're all pretty closely related, and often get updated at the same time, so I probably would have gone with `useReducer` and some nicely semantic actions instead. 

While performance shouldn't be much of a concern here, it is still important to note that each state update can (does?) lead to its own render, so depending on the semantics (I don't think it's a problem in this particular case), you could end up with invalid intermediate renders where only part of the state was updated.

Reply:
I'll do a pass that removes the CSS-specific bits because that will be redundant very soon.

We can see if these seem more reasonable then…

Reply:
I think as of f76ec5ee9e52ed7e5977234793775b2852a24eb8 this is much more reasonable, but I'd love to hear what you think 🙇